src/groth16/aggregate/commit.rs

```diff
@@ -0,0 +1,152 @@
+/// TODO global commitment doc and usage
+use ff::{Field, PrimeField};
+use groupy::{CurveAffine, CurveProjective};
+use rayon::prelude::*;
+
+use crate::bls::Engine;
+use crate::groth16::aggregate::inner_product;
+
+/// Key is a generic commitment key that is instanciated with g and h as basis,
+/// and a and b as powers.
+#[derive(Clone, Debug)]
+pub struct Key<G: CurveAffine> {
+    /// Exponent is a
+    pub a: Vec<G>,
+    /// Exponent is b
+    pub b: Vec<G>,
+}
+/// VKey is a commitment key used by the "single" commitment on G1 values as
+/// well as in the "pair" commtitment.
+/// It contains $\{h^a^i\}_{i=1}^n$ and $\{h^b^i\}_{i=1}^n$
+pub type VKey<E> = Key<<E as Engine>::G2Affine>;
+
+/// WKey is a commitment key used by the "pair" commitment. Note the sequence of
+/// powers starts at $n$ already.
+/// It contains $\{g^{a^{n+i}}\}_{i=1}^n$ and $\{g^{b^{n+i}}\}_{i=1}^n$
+pub type WKey<E> = Key<<E as Engine>::G1Affine>;
+
+impl<G> Key<G>
+where
+    G: CurveAffine,
+{
+    /// correct_len returns true if both commitment keys have the same size as
+    /// the argument. It is necessary for the IPP scheme to work that commitment
+    /// key have the exact same number of arguments as the number of proofs to
+    /// aggregate.
+    pub fn correct_len(&self, n: usize) -> bool {
```

Comment:
Usually methods that return bools are named sth like `is_` or `has_`, this sounds more like it returns the length, not checks for sth